### ISSUE TITLE
It's not possible to set withCredentials false

### ISSUE THREAD

**Issue by fabiopaiva** [axios/axios#1837]
#### Summary

I would like to set the property `withCredentials` as `false` but it's [checking](https://github.com/axios/axios/blob/master/lib/adapters/xhr.js#L126) a falsy boolean to set it:

```
// Add withCredentials to request if needed
    if (config.withCredentials) {
      request.withCredentials = true;
    }
```

My problem is using React Native, which sets the default value of `withCredentials` as `true`. [Here](https://github.com/facebook/react-native/blob/master/Libraries/Network/XMLHttpRequest.js#L125)

Then, axis will ignore if I set `withCredentials` `false` on my request.
 
I would suggest something like this:

```
// Add withCredentials to request if needed
    if (typeof config.withCredentials !== 'undefined') {
      request.withCredentials = !!config.withCredentials;
    }
```

**Comment by fabiopaiva** [comment]
https://github.com/facebook/react-native/issues/21852

**Comment by ikhilko** [comment]
It's fixed in https://github.com/axios/axios/pull/1852

**Comment by yasuf** [comment]
sorry if this is sort of an obvious question, did you try setting it to false on the config you send to axios? even if `react-native` sets it to true as a default, if you change it to false, what's inside the conditional should not be run right? and credentials wouldn't be used in that case, let me know if I'm missing something here!

**Comment by yasuf** [comment]
sorry, I got it, react native sets it to true at the XMLHttpRequest level

**Comment by chinesedfan** [comment]
Solved by #2582.
